tests/test_statistics_store_isolation.py

```
@@ -0,0 +1,66 @@
+    store.append_snapshot("shadow", "run-a", "OBSERVING", kpi, [])
+    with sqlite3.connect(store.path) as con:
+        stored = json.loads(con.execute("SELECT kpi_json FROM snapshots").fetchone()[0])
+    assert stored == {"realized_pnl": 1.5}
+
+
+def test_snapshot_retention_caps_rows_and_reclaims_space(tmp_path: Path):
+    """Without a retention cap a long run accumulates thousands of
+    half-megabyte rows and the stats DB grows unbounded."""
+    store = StatisticsStore.create(tmp_path / "data", "20260829_120005", "run-a")
+    fat_kpi = {"by_market": {"mkt": {"data": "x" * 100_000}}}
+    for i in range(60):
```

Ruff (0.16.2)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)